### ISSUE TITLE
Conversion from "BigInt" to "number" is not allowed (not the exponential issue)

### ISSUE THREAD

**Issue by khoanguyen-yang** [paulmillr/noble-hashes#40]
I got this error inside `@noble/hashes`. Got to know the same error message but for the exponential operator but I do not think it has anything relevant to mine.

Really appreciate any advice.

![image](https://user-images.githubusercontent.com/84629610/205860519-b4354e8b-1201-4303-ab42-47bfabb81917.png)

**Comment by paulmillr** [comment]
You need to provide a code that would reproduce the issue.

**Comment by khoanguyen-yang** [comment]
I did not directly use your lib. It is used by another one which is `@solana/web3.js`. `sha512` is used specifically

This is what I got from the stacktrace. 

Image 1: Inside `sha512.js` , the method `split` method of `u64` is called
Image 2: The `split` method then uses `fromBig` which has the `BigInt` conversion to `Number` which throws the above error.

I am using this on React Native so I think this error might not relevant to your repo. But if you have any advice, I would really appreciate.

<img width="862" alt="image" src="https://user-images.githubusercontent.com/84629610/205866632-6a79c0bc-347a-44a1-979f-caa0967e96d3.png">
<img width="725" alt="image" src="https://user-images.githubusercontent.com/84629610/205866772-093f7f07-aa24-4f7b-aa51-3df1fec3b77d.png">
